src/platform/plugins/shared/workflows_management/public/widgets/workflow_yaml_editor/ui/step_action_options/copy_elastic_search_devtools_option.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React, { useCallback } from 'react';
+import { EuiContextMenuItem } from '@elastic/eui';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import type { CoreStart } from '@kbn/core/public';
+import type { ElasticsearchGraphNode } from '@kbn/workflows/graph/types';
+import { useSelector } from 'react-redux';
+import { selectFocusedStepInfo, selectWorkflowGraph } from '../../lib/store';
+import { getElasticsearchRequestInfo } from '../../lib/elasticsearch_step_utils';
+
+export interface CopyElasticSearchDevToolsOptionProps {
+  onClick: () => void;
+}
+
+export const CopyElasticSearchDevToolsOption: React.FC<CopyElasticSearchDevToolsOptionProps> = ({
+  onClick,
+}) => {
+  const workflowGraph = useSelector(selectWorkflowGraph);
+  const focusedStepInfo = useSelector(selectFocusedStepInfo);
+  const {
+    services: { notifications },
+  } = useKibana<CoreStart>();
+
+  function generateConsoleFormat(
+    requestInfo: { method: string; url: string; data?: string[] },
+    withParams: Record<string, any>
+  ): string {
+    const lines = [`${requestInfo.method} ${requestInfo.url}`];
+
+    if (requestInfo.data && requestInfo.data.length > 0) {
+      lines.push(...requestInfo.data);
+    }
+
+    return lines.join('\n');
+  }
+
+  const copy = useCallback(async () => {
+    if (!focusedStepInfo) {
+      return;
+    }
+
+    try {
+      const stepGraph = workflowGraph?.getStepGraph(focusedStepInfo?.stepId);
+      const elasticSearchNode = stepGraph?.getAllNodes()[0] as ElasticsearchGraphNode;
```

Comment:
NIT: It would be better to actually check that it's an `ElasticsearchGraphNode` than just casting it.